ksql-functional-tests/src/main/java/io/confluent/ksql/test/tools/KsqlTestingTool.java

```diff
@@ -0,0 +1,119 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.test.tools;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.confluent.ksql.function.TestFunctionRegistry;
+import io.confluent.ksql.json.JsonMapper;
+import io.confluent.ksql.test.model.QttTestFile;
+import io.confluent.ksql.test.model.TestCaseNode;
+import io.confluent.ksql.test.tools.command.TestOptions;
+import io.confluent.ksql.util.Pair;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class KsqlTestingTool {
+
+  private KsqlTestingTool() {
+
+  }
+
+  private static final ObjectMapper OBJECT_MAPPER = JsonMapper.INSTANCE.mapper;
+
+  private static int totalNumberOfTests;
+
+  public static void main(final String[] args) throws IOException {
+    loadAndRunTests(args);
+  }
+
+  static void loadAndRunTests(final String[] args) {
+    totalNumberOfTests = 0;
+    final List<String> passedTests = new ArrayList<>();
+    final List<Pair<String, String>> failedTests = new ArrayList<>();
+    try {
+
+      final TestOptions testOptions = TestOptions.parse(args);
+
+      if (testOptions == null) {
+        System.err.println("Invalid parameter.");
+        System.err.println("Usage: ksql-testing-tool /path/to/the/test/file.json");
+        return;
+      }
+
+      final QttTestFile qttTestFile = OBJECT_MAPPER.readValue(
+          new File(testOptions.getTestFile()), QttTestFile.class);
+      for (final TestCaseNode testCaseNode: qttTestFile.tests) {
+        final List<TestCase> testCases = testCaseNode.buildTests(
+            new File(testOptions.getTestFile()).toPath(),
+            TestFunctionRegistry.INSTANCE.get());
+        for (final TestCase testCase: testCases) {
+          executeTestCase(
+              testCase,
+              new TestExecutor(),
+              passedTests,
+              failedTests);
+        }
+      }
+      printResults(totalNumberOfTests, passedTests, failedTests);
+    } catch (final Exception e) {
+      System.err.println("Failed to start KSQL testing tool: " + e.getMessage());
+    }
+  }
+
+  static void executeTestCase(
+      final TestCase testCase,
+      final TestExecutor testExecutor,
+      final List<String> passedTests,
+      final List<Pair<String, String>> failedTests
+  ) {
+    try {
+
+      System.out.println(" >>> Running test: " + testCase.getName());
+      testExecutor.buildAndExecuteQuery(testCase);
+      System.out.println("\t >>> Test " + testCase.getName() + " passed!");
+      passedTests.add(testCase.getName());
+      totalNumberOfTests ++;
+    } catch (final Exception e) {
+      e.printStackTrace();
+      System.err.println("\t>>>>> Test " + testCase.getName() + " failed: " + e.getMessage());
```

Comment:
In the second example I put above (copy-pasted below for convenience), this line is not getting executed, I think because the AssertionError isn't being properly passed up from a lower try-catch, e.g., the one in `TestExecutor#buildAndExecuteQuery()`. We should figure out where the exception is getting swallowed and fix it so this line prints. (Something is wrong with my IntelliJ configuration and I can't run the debugger right now... I can help you debug once I figure this out.)

```
Victoria-Xias-MBP13:ksql victoria.xia$ ./bin/ksql-testing-tool ksql-functional-tests/src/test/resources/query-validation-tests/hopping-windows.json
 >>> Running test: hopping-windows - max hopping
Exception in thread "main" java.lang.AssertionError:
Expected: <7>
     but: was <18>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at io.confluent.ksql.test.tools.TestCase.validateTopicData(TestCase.java:396)
	at io.confluent.ksql.test.tools.TestCase.lambda$verifyOutputTopics$8(TestCase.java:385)
	at java.util.HashMap$KeySet.forEach(HashMap.java:933)
	at io.confluent.ksql.test.tools.TestCase.verifyOutputTopics(TestCase.java:385)
	at io.confluent.ksql.test.tools.TestExecutor.buildAndExecuteQuery(TestExecutor.java:109)
	at io.confluent.ksql.test.tools.KsqlTestingTool.executeTestCase(KsqlTestingTool.java:87)
	at io.confluent.ksql.test.tools.KsqlTestingTool.loadAndRunTests(KsqlTestingTool.java:65)
	at io.confluent.ksql.test.tools.KsqlTestingTool.main(KsqlTestingTool.java:41)
```

Reply:
Another thing worth noting: the file `hopping-windows.json` has multiple tests in it, but because of this failure from the first test, subsequent tests in the file weren't run. This makes me think that instead of an exception being swallowed (as I'd previously guessed), it's actually an error that's not being caught.

Reply:
This one is fixed now. There was a `printStackTrace()` statement that I hadn't removed.

Reply:
Hmm... I still seem to be hitting it. For example:
```
Victoria-Xias-MBP13:ksql victoria.xia$ ./bin/ksql-testing-tool ksql-functional-tests/src/test/resources/query-validation-tests/histogram.json
 >>> Running test: histogram - histogram string - AVRO
	>>>>> Test histogram - histogram string - AVRO failed: java.util.LinkedHashMap cannot be cast to java.util.List
 >>> Running test: histogram - histogram string - JSON
	>>>>> Test histogram - histogram string - JSON failed: Expected <0, {"ID":0,"COUNTS":{"foo":1,"bar":1}}> with timestamp=0 but was <0, {"ID":0,"COUNTS":{"bar":1,"foo":1}}> with timestamp=0
 >>> Running test: histogram - histogram on a table - AVRO
	>>>>> Test histogram - histogram on a table - AVRO failed: java.util.LinkedHashMap cannot be cast to java.util.List
 >>> Running test: histogram - histogram on a table - JSON
Exception in thread "main" java.lang.AssertionError:
Expected: <6>
     but: was <7>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at io.confluent.ksql.test.tools.TestCase.validateTopicData(TestCase.java:437)
	at io.confluent.ksql.test.tools.TestCase.lambda$verifyOutputTopics$8(TestCase.java:426)
	at java.util.HashMap$KeySet.forEach(HashMap.java:933)
	at io.confluent.ksql.test.tools.TestCase.verifyOutputTopics(TestCase.java:426)
	at io.confluent.ksql.test.tools.TestExecutor.buildAndExecuteQuery(TestExecutor.java:109)
	at io.confluent.ksql.test.tools.KsqlTestingTool.executeTestCase(KsqlTestingTool.java:86)
	at io.confluent.ksql.test.tools.KsqlTestingTool.loadAndRunTests(KsqlTestingTool.java:65)
	at io.confluent.ksql.test.tools.KsqlTestingTool.main(KsqlTestingTool.java:41)
```
Because the fourth test threw an AssertionError, the rest of the tests in the file weren't run and no summary was printed.